predicates/forwarded/forwarded_test.go

```diff
@@ -0,0 +1,308 @@
+package forwarded
+
+import (
+	"net/http"
+	"net/url"
+	"testing"
+)
+
+type request struct {
+	url     string
+	headers http.Header
+}
+
+func TestForwardedHost(t *testing.T) {
+	testCases := []struct {
+		msg     string
+		host    string
+		r       request
+		matches bool
+		isError bool
+	}{{
+		msg:     "Empty host should fail",
+		host:    "",
+		r:       request{},
+		matches: false,
+		isError: true,
+	}, {
+		msg:     "Empty Forwarded Header should not match",
+		host:    "example.com",
```

Comment:
IMO we should be more precise in the test examples as they also serve as a documentation. 
`example.com` regexp also matches `host=example.comma.org` as far as I can tell.

Reply:
Great point!

Reply:
I think it should be `^example\.com$` (note escaped dot) everywhere

Reply:
Done!